x/axelarnet/message_handler.go

```diff
@@ -191,15 +268,22 @@ func handleMessage(ctx sdk.Context, n types.Nexus, sourceAddress nexus.CrossChai
 }
 
 func handleMessageWithToken(ctx sdk.Context, n types.Nexus, b types.BankKeeper, sourceAddress nexus.CrossChainAddress, msg Message, asset keeper.Coin) error {
-	destChain := funcs.MustOk(n.GetChain(ctx, nexus.ChainName(msg.DestinationChain)))
+	id := n.GenerateMessageID(ctx)
 
+	fee, err := deductFee(ctx, b, msg.Fee, funcs.Must(asset.GetOriginalDenom()), id)
+	if err != nil {
+		return err
+	}
+	// subtract fee from transfer value
+	asset.Amount = asset.Amount.Sub(fee)
```

Comment:
you're already deducting fees, why this extra manual step?

Reply:
I can pass the token pointer in, and deduct fees inside the function. prefer to be explicit though. 
how about change name to `chargeFee`?

Reply:
I'd also be ok with passing `asset` in and returning it after deducting the fee, as opposed to passing the pointer.

Reply:
ok, let's go with pass asset and return after fee asset, `defuctFee` name matches the behaviour